src/core/Akka/Actor/TypedActor.cs

```diff
@@ -34,7 +35,10 @@ public abstract class TypedActor : ActorBase
         /// <returns>TBD</returns>
         protected sealed override bool Receive(object message)
         {
-            MethodInfo method = GetType().GetMethod("Handle", new[] {message.GetType()});
+            Type[] msgTypeArray = new[] {message.GetType()};
+            Type actorType = GetType();
+            MethodInfo method = actorType.GetMethod("Handle", msgTypeArray) ??
+                                actorType.GetMethod($"Akka.Actor.IHandle<{msgTypeArray[0].FullName}>.Handle", BindingFlags.NonPublic | BindingFlags.Instance);
```

Comment:
I would try something like this:
![image](https://cloud.githubusercontent.com/assets/4236651/25339176/102b5d76-2902-11e7-9f13-97a7f3543566.png)